laravel-api/api/User/Services/UserPasswordService.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Api\User\Services;
+
+use Illuminate\Foundation\Application;
+use Infrastructure\Auth\Exceptions\InvalidCredentialsException;
+use Api\User\Repositories\UserRepository;
+use Api\Domain\Repositories\DomainRepository;
+use Api\User\Repositories\Contact_emailRepository;
+use Illuminate\Support\Facades\Password;
+use Illuminate\Support\Str;
+use Illuminate\Auth\Events\PasswordReset;
+use Api\User\Exceptions\UserDisabledException;
+use Webpatser\Uuid\Uuid;
+
+class UserPasswordService
+{
+
+    private $request;
+
+    private $userRepository;
+
+    private $domainRepository;
+
+    private $contact_emailRepository;
+
+    public function __construct(
+      Application $app,
+      UserRepository $userRepository,
+      DomainRepository $domainRepository,
+      Contact_emailRepository $contact_emailRepository
+    )
+    {
+        $this->userRepository = $userRepository;
+        $this->domainRepository = $domainRepository;
+        $this->contact_emailRepository = $contact_emailRepository;
+
+        $this->request = $app->make('request');
+    }
+
+    public function generateResetToken($email, $domain_name)
+    {
+        $user = $this->getUserCredentials($email, $domain_name);
+        $userCredentials = array_merge(
+            $this->request->only('user_email'), $user->toArray()
+        );
+        $status = Password::sendResetLink($userCredentials);
+
+        return [
+            'username' => $userCredentials['username'],
+            'domain_uuid' => $userCredentials['domain_uuid'],
+        ];
+    }
+
+    public function resetPassword($email) 
+    {
+        $user = $this->getUserCredentials($email);
+        $userCredentials = array_merge(
+            $this->request->only(
+                'user_email','password', 'password_confirmation', 'token'
+            ),
+            $user->toArray()
+        );
+
+        Password::reset(
+            $userCredentials,
+            function ($user, $password) {
+                $data['salt'] = Uuid::generate();
+                $data['password'] = md5($data['salt'] . $password);
+                $user->fill($data);
```

Comment:
Probably $user->fill will not work as User model should have at least `salt` guarded.